File: app/src/main/java/com/zxzhu/show/units/base/BaseFragment.java
package com.zxzhu.show.units.base;

import android.databinding.DataBindingUtil;
import android.databinding.ViewDataBinding;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

/**
 * Created by zxzhu on 2017/8/2.
 */

public abstract class BaseFragment extends Fragment {

    private View mView;
    private LayoutInflater inflater;
    private ViewGroup container;
    private ViewDataBinding binding;
    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        mView = inflater.inflate(getResourceId(),container,false);
        binding = DataBindingUtil.inflate(inflater, getResourceId(), container, false);
        this.inflater = inflater;
        this.container = container;
        initData();
        return binding.getRoot();
    }

    //做数据或其他初始化的方法
    protected abstract void initData();

    //获取当前fragment的view的方法
    protected abstract int getResourceId();

    public void toast(String str) {
        Toast.makeText(getActivity(),str,Toast.LENGTH_SHORT).show();
    }



    public LayoutInflater getInflater(){
        return inflater;
    }

    public ViewGroup getViewGroup(){
        return container;
    }

    //获取对应ViewDataBinding
    public <T extends ViewDataBinding>T getDataBinding() {
        return (T)binding;
    }

    //View快捷绑定id的方法
    public <T extends View> T $(int id) {
        return (T) mView.findViewById(id);
    }
}

File: app/src/main/java/com/zxzhu/show/units/base/BaseFragment.kt
package com.zxzhu.show.units.base

import android.databinding.DataBindingUtil
import android.databinding.ViewDataBinding
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast

/**
 * Created by zxzhu on 2017/8/2.
 */

@Suppress("UNCHECKED_CAST")
abstract class BaseFragment : Fragment() {

    private var mView: View? = null
    var inflater: LayoutInflater? = null
        private set
    var viewGroup: ViewGroup? = null
        private set
    private var binding: ViewDataBinding? = null

    //获取当前fragment的view的方法
    protected abstract val resourceId: Int

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        mView = inflater.inflate(resourceId, container, false)
        binding = DataBindingUtil.inflate(inflater, resourceId, container, false)
        this.inflater = inflater
        this.viewGroup = container
        initData()
        return binding!!.root
    }

    //做数据或其他初始化的方法
    protected abstract fun initData()

    fun toast(str: String) {
        Toast.makeText(activity, str, Toast.LENGTH_SHORT).show()
    }

    //获取对应ViewDataBinding
    fun <T : ViewDataBinding> getDataBinding(): T? {
        return binding as T?
    }

    //View快捷绑定id的方法
    fun <T : View> `$`(id: Int): T {
        return mView!!.findViewById<View>(id) as T
    }
}
